packages/grid/_modules_/grid/hooks/features/filter/gridFilterUtils.ts

```diff
@@ -54,6 +37,70 @@ export const cleanFilterItem = (item: GridFilterItem, apiRef: GridApiRef) => {
   return cleanItem;
 };
 
+const filterModelDisableMultiColumnsFilteringWarning = buildWarning([
+  'MUI: The `filterModel` can only contain a single item when `prop.disableMultipleColumnsFiltering` is set to `true`.',
+  'If you are using the `DataGrid` community version, this property is always `true`.',
+]);
+
+const filterModelMissingItemIdWarning = buildWarning(
+  "MUI: The 'id' field is required on `filterModel.items` when you use multiple filters.",
+);
+
+export const sanitizeFilterModel = (
+  model: GridFilterModel,
+  disableMultipleColumnsFiltering: boolean,
+  apiRef: GridApiRef,
+) => {
+  if (model.items.length > 1) {
+    if (disableMultipleColumnsFiltering) {
+      filterModelDisableMultiColumnsFilteringWarning();
+
+      return {
+        ...model,
+        items: [model.items[0]],
+      };
+    }
+
+    let hasItemsWithoutIds = false;
+    let hasItemWithoutOperator = false;
+
+    for (let i = 0; i < model.items.length; i += 1) {
```

Comment:
If performance is one concern we could stop iterating once `hasItemsWithoutIds` or `hasItemWithoutOperator` is true. Otherwise I wouldn't  mind if we used `Array.some` to compute theses variables. In practice there won't be more than a few filters applied at the same time.

Reply:
Yes performance is not a concern at all here, I don't imagine a grid with like 10 000 filter items.